packages/eslint-plugin/src/rules/keyword-spacing.ts

```diff
@@ -56,6 +56,11 @@ export default util.createRule<Options, MessageIds>({
       ): void {
         const typeToken = sourceCode.getFirstToken(node, { skip: 1 })!;
         const punctuatorToken = sourceCode.getTokenAfter(typeToken)!;
+        if (
+          node.specifiers?.[0]?.type === AST_NODE_TYPES.ImportDefaultSpecifier
+        ) {
+          return;
+        }
```

Comment:
I'm confused. The preview deploy's playground still shows an unexpected space for one of the three issue's sample cases: https://deploy-preview-6095--typescript-eslint.netlify.app/play/#ts=4.9.3&sourceType=module&code=FASwtgDg9gTgLgAjgTwgUwQbwQZQIYBuaAJgIoCuaMIaAzggL4IBmMUYCARAHQD0+RMpWp1uAK1qcA3KEixEKdAgAWcOBABMLNh04A7KMTQAuVeo3TZ0eElQYzEbey4GjptREtA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Aa0UIHcB7aACa1kxAIZlmAc3RgA2uGw5uo6E0gAaBYqyLskAEaIAZv0Qz8cRJt17RR-Imgyjo+MitbdkXgDdHVARQZHRscRDczDBDQnDsHJwwLWERPGwBfaxioaER8WFVg1NDIOMdzSyLFDMq9C0JCrNt7MsSKxurGyDJRfDIACwbGqFKEsBcImogOrK7Rd0HOkfLkybBpmK7eJmR8BZml1pX2zI38Puhebj2Ng7AklOPVyCRdqNXY5tH71fXigV5rsVbt9Hp1eEZATYSp9lg8sr8oXNIV5gW14SdikZmK54PU3kNhjDDnCYgivEYLgBbZGKaHxWE-DFQyiU4j8V6Yd6E+nExlPRAADzZ0A50RuRLuaNJTK8gRcCFFXLpLUlR3hNTSRU1igAugpNWkgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA. Am I missing something?

```ts
import type { SavedQueries } from "./SavedQueries.js";
```

Reply:
I thought draft means you won't review it (Sorry, I guess it meant something else to me).

I opened it for review because I can't think of a way to solve it without adding an override option for the `type` keyword or should it use the existing option for the `import` keyword.

Reply:
Oh I just didn't realize there was a question associated with this! Sorry, i misinterpreted the discussion. Will think on it 🙂

Reply:
Or, actually, definitely yes. Since the `type` option doesn't exist in ESLint core and is TypeScript-specific, we should add in a feature for it. Yes please 😄

Reply:
Any example I can look at how to do that? The types seem looser than the schema validation (typed as `overrides?: Record<string, { before?: boolean; after?: boolean;}>` but adding `type: { after: true }` to object breaks the schema validation